Sport Shop is a store specialized in sportswear. Its flagship product is running shoes. The running shoes that they offer usually divide them into two groups: white running shoes (which are totally white) and colored shoes (which have two or more colors). The Chief of Inventories and the Marketing Team want to know the preferences of their customers regarding said products, for that they ask us to answer the following questions:

- What is the probability that a person prefers white running shoes?

- What is the probability the person is male AND prefers color running shoes?

- What is the probability the person is female OR prefers color running shoes?

In [1]:
#Let's import the dataset that contains the pertinent information. We have taken as a study the last hundred purchases made by our customers.

In [ ]:
import pandas as pd

In [ ]:
dataset = pd.read_excel('C:/Users/USUARIO/Documents/CRISTINA/DATA SCIENCE/PORTFOLIO/SPORT SHOP/SPORT SHOP.xlsx')

In [ ]:
#Let's know the characteristics of our dataset

In [ ]:
dataset.head()

GENDER WHITE COLORS
0  FEMALE    NO    YES
1    MALE   YES    YES
2  FEMALE   YES    YES
3    MALE   YES     NO
4  FEMALE   YES     NO

In [ ]:
dataset.shape

(100, 3)

In [ ]:
#Let's find out how many women and how many men bought white running shoes. For this we will make a contingency table.

In [ ]:
contingency_white = pd.crosstab(dataset['GENDER'], dataset['WHITE'])

In [ ]:
contingency_white

WHITE   NO  YES
GENDER         
FEMALE  20   25
MALE    10   45

In [ ]:
#Let's find out how many women and how many men bought colors running shoes. For this we will also make a contingency table.

In [ ]:
contingency_colors = pd.crosstab(dataset['GENDER'], dataset['COLORS'])

In [ ]:
contingency_colors

COLORS  NO  YES
GENDER         
FEMALE  25   20
MALE    45   10

In [ ]:
#Let's check all the customers under study.

In [ ]:
contingency_white.sum(axis = 1)

GENDER
FEMALE    45
MALE      55
dtype: int64

In [ ]:
contingency_colors.sum(axis = 1)

GENDER
FEMALE    45
MALE      55
dtype: int64

In [ ]:
contingency_white.sum(axis = 0)

WHITE
NO     30
YES    70
dtype: int64

In [ ]:
#Total number of customers

In [ ]:
30+70

100

In [ ]:
contingency_colors.sum(axis = 0)

COLORS
NO     70
YES    30
dtype: int64

In [ ]:
#Total number of clients

In [ ]:
70+30

100

In [ ]:
from IPython.display import display, Math, Latex

In [ ]:
#Let's begin to establish our probability formulas in order to solve the questions asked by the work team.

What is the probability that a person prefers white running shoes?

In [ ]:
display(Math(r'P(white\ running\ shoes) = \frac{70}{100}'))

<IPython.core.display.Math object>

In [ ]:
70/100

0.7

What is the probability the person is male AND prefers color running shoes?

In [ ]:
display(Math(r'P(Male \cap colors\ running\ shoes) = \frac{10}{100}'))

<IPython.core.display.Math object>

In [ ]:
10/100

0.1

What is the probability the person is female OR prefers color running shoes?

In [ ]:
# P(Female U colors running shoes) = P(Female) + P(color runnning shoes) – P(Female ∩ color running shoes)

In [ ]:
45/100

0.45

In [ ]:
30/100

0.3

In [ ]:
20/100

0.2

In [ ]:
0.45+0.3-0.2

0.55

We have reached the following conclusions:

- What is the probability that a person prefers white running shoes?

P=0.7, indicates that it exists a 70% probability that a customer prefers to buy white running shoes. Which means that this event is very likely to occur.

- What is the probability the person is male AND prefers color running shoes?

P=0.1, indicates that it exits a 10% probability that a customer is male and prefers to buy colored running shoes. If P=0.1, it is a probability very close to 0, which indicates that the event is very unlikely to occur.

- What is the probability the person is female OR prefers color running shoes?

P=0.55, indicates that it exists a 55% probability that a customer is female or prefers to buy color running shoes. If P=0.55 it is a probability very close to 1/2, which indicates that it is equally likely that this event occurs or not.